telethon/telegram_bare_client.py

```diff
@@ -198,20 +209,20 @@ def __init__(self, session, api_id, api_hash,
             # another data center and this would raise UserMigrateError)
             # to also assert whether the user is logged in or not.
             self._user_connected = True
-            if _sync_updates and not _cdn:
+            if _sync_updates and not _cdn and not self._authorized:
                 try:
                     await self.sync_updates()
                     self._set_connected_and_authorized()
                 except UnauthorizedError:
-                    self._authorized = False
+                    pass
```

Comment:
Leftover?

Reply:
There is no exceptions in `_set_connected_and_authorized` expected, but if we get `UnauthorizedError` on `self.sync_updates()`, `self._authorized` remains `False`.